### PR TITLE
Complete task 1

### PR DESCRIPTION
**Pull request by dgabka**

### CODE REVIEW

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const Button = props => (
+    <button
+        className={props.isDelete ? 'delete-btn' : 'default-btn'}
+        onClick={props.clicked}>
+        {props.children}
+    </button>
+);
+
+Button.propTypes = {
+    clicked: PropTypes.func.isRequired,
```
Comment (prz-szo) [inline src/components/Button/Button.js]: `clicked` doesn't sound like the name for the callback. `onClick` is better :D

File: src/components/Posts/Posts.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                />);
+
+		return (
+			<div className={'posts'}>
+				<div className={'posts-menu'}>
+					<Search filter={this.handleFilter}/>
+					<Button clicked={this.handleAddPost}>Add Post</Button>
+				</div>
+				<div>{postList}</div>
+				{this.state.posts.length !== this.state.displayedPosts.length ?
```
Comment (prz-szo) [inline src/components/Posts/Posts.js]: To extract to separate method. `renderLoadMore()` and also `shouldRenderLoadMore()`

Comment (prz-szo) [reply src/components/Posts/Posts.js]: In cases you want to do something only if the condition was met - use short circuit evaluation. 
`{ [condition] && [some action e.g. render <div/>] }` - In this case in the braces we will get `false` or `<div/>` element, as if the condition is false the second operand is not evaluated

Comment (prz-szo) [reply src/components/Posts/Posts.js]: On the other hand there is good convention to write ` ? : ` operator:
```
[condition]
    ?  [action on true]
    :  [action on false]
```

File: src/components/Posts/Posts.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                posts: response.data,
+                displayedPosts: response.data.slice(0, 4),
+            }));
+    }
+
+    render() {
+        const postList = this.state.displayedPosts
+            .filter(post => post.title.includes(this.state.filterValue) || post.body.includes(this.state.filterValue))
+            .map(post =>
+                <PostItem
```
Comment (prz-szo) [inline src/components/Posts/Posts.js]: You can even extract rendering one `PostItem` element :D

File: src/components/Posts/PostItem/PostItem.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            <Button clicked={props.open}>Open</Button>
+            <Button clicked={props.delete} isDelete>Delete</Button>
+        </div>
+    </div>
+);
+
+PostItem.propTypes = {
+    title: PropTypes.string.isRequired,
+    body: PropTypes.string.isRequired,
+    delete: PropTypes.func.isRequired
```
Comment (prz-szo) [inline src/components/Posts/PostItem/PostItem.js]: We usually name callbacks with the preposition `on` - `onDelete`

File: src/components/Posts/PostItem/PostItem.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            <p>{props.body}</p>
+        </div>
+        <div className={'post-item-buttons'}>
+            <Button clicked={props.open}>Open</Button>
+            <Button clicked={props.delete} isDelete>Delete</Button>
+        </div>
+    </div>
+);
+
+PostItem.propTypes = {
```
Comment (prz-szo) [inline src/components/Posts/PostItem/PostItem.js]: The is no `open: PropTypes.func.isRequired` 🍶

Comment (prz-szo) [reply src/components/Posts/PostItem/PostItem.js]: Or rather `onOpen` 
```
  |
 \|/
```

File: src/components/Posts/PostItem/PostItem.test.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PostItem from './PostItem';
+
+it('renders without crashing', () => {
```
[the remark this answers is not in the document]
Comment (dgabka) [reply src/components/Posts/PostItem/PostItem.test.js]: Yes, I was going to write tests as I go, but then decided to first finish the app and then focus on learning testing. Deleted it for now.

File: src/components/Posts/Posts.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                    title={post.title}
+                    body={post.body}
+                    key={post.id}
+                    delete={() => this.handleDeletePost(post.id)}
+                    open={() => this.handleOpenPost(post.id)}
+                />);
+
+		return (
```
[the remark this answers is not in the document]
Comment (dgabka) [reply src/components/Posts/Posts.js]: I'm used to tabs, so I've switched to tabs, but forgot to refactor all files accordingly 😭 
I'm going to try out 2 spaces now

**Comment by dgabka** [comment]
@prz-szo I've just pushed the fixes, so whenever you're ready

**Pull request merged**